unit/testing-utils/require_goto_statements.cpp

```diff
@@ -55,18 +62,59 @@ require_goto_statements::find_struct_component_assignments(
       if(code_assign.lhs().id() == ID_member)
       {
         const auto &member_expr = to_member_expr(code_assign.lhs());
-        const auto &symbol = member_expr.symbol();
 
-        if(
-          symbol.get_identifier() == structure_name &&
-          member_expr.get_component_name() == component_name)
+        if(superclass_name.has_value())
         {
-          component_assignments.push_back(code_assign);
+          const auto &superclass_compound = member_expr.compound();
+
+          if(
+            member_expr.get_component_name() == component_name &&
+            superclass_compound.id() == ID_member)
+          {
+            const auto &superclass_member_expr =
+              to_member_expr(superclass_compound);
+
+            if(
+              superclass_member_expr.get_component_name() ==
```

Comment:
This condition is hard to follow (partly down to line breaks) consider breaking out the two conditions into intermediate variables